torch/distributed/launch.py

```diff
@@ -0,0 +1,172 @@
+"""
+torch.distributed.launch is a helper module that spawns up multiple distributed
+training processes on each of the training nodes.
+
+The utility can be used for single-node distributed training, in which
+each distributed process will be operating on a single GPU. This can achieve
+well-improved single-node training performance. It can also be used in
+multi-node distributed training, by spawning up multiple processes on each node
+for well-improved multi-node distributed training performance as well.
+This will especially be benefitial for systems with multiple Infiniband
+interfaces since all of them can be utilized for aggregated communication
+bandwidth.
+
+In both cases, this utilily will launch a given number of processes
+per node (nproc_per_node, which defaults to the number of GPUs on the node).
+This number needs to be less or euqal to the number of GPUs on the current
+system, and each process will be operating on a single GPU from GPU 0 to
+GPU nproc_per_node - 1.
+
+How to use this module:
+
+    (1) Single Node multi-proc distributed training:
+        python -m torch.distributed.launch YOUR_TRAINING_SCRIPT.py (--arg1
+        --arg2 --arg3 and all other arguments of your training script)
+
+    (2) Multi Node multi-proc distributed training: (e.g. two nodes)
+
+    NODE1: (IP: 192.168.1.1, and has a free port: 1234)
+        python -m torch.distributed.launch --nproc_per_node=NUM_GPUS_YOU_HAVE
+        --num_node=2 --rank_node=0 --master_addr="192.168.1.1"
+        --master_port=1234 YOUR_TRAINING_SCRIPT.py (--arg1 --arg2 --arg3 and
+        all other arguments of your training script)
+    NODE2:
+        python -m torch.distributed.launch --nproc_per_node=NUM_GPUS_YOU_HAVE
+        --num_node=2 --rank_node=1 --master_addr="192.168.1.1"
+        --master_port=1234 YOUR_TRAINING_SCRIPT.py (--arg1 --arg2 --arg3 and
+        all other arguments of your training script)
```

Comment:
We need an example that shows how to use other initialization methods. In particular, we should say that the arguments are optional, and if you use them, then you should use the `env://` init method

Reply:
I disagree. Other initialization methods are even simpler because you *don't need this information* (it is detected automatically). So there's no reason to make those arguments required